custom/icds_reports/ucr/expressions.py

```diff
@@ -133,7 +139,11 @@ def __call__(self, item, context=None):
 
     def _get_forms(self, case_id, from_date, to_date, context):
         domain = context.root_doc['domain']
-        cache_hash = "{}{}{}".format(from_date.toordinal(), to_date.toordinal(), self.count)
+        cache_hash = "{}".format(self.count)
+        if from_date:
+            cache_hash += "{}".format(from_date.toordinal())
+        if to_date:
+            cache_hash += "{}".format(to_date.toordinal())
```

Comment:
I think you might need `None` values for from_date and to_date to be represented in the hash. Otherwise, an expression with "from:2017-03-28 to:None" and "from:None to:2017-03-28" will have the same hash.

Speaking of which, would it be preferable to just use a longer tuple that includes each argument as the `cache_key`?